pkg/controller/controller_instance.go

```diff
@@ -250,19 +297,30 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 
 	glog.V(4).Infof("Adding/Updating Instance %v/%v", instance.Namespace, instance.Name)
 
+	// we will definitely update the instance's status - make a deep copy now
+	// for use later in this method.
+	clone, err := api.Scheme.DeepCopy(instance)
```

Comment:
Any time the status is changed, it should be changed on a deep copy, so that you do not mutate what's in the cache.  It has nothing to do with multiple fields on the status being changed.  The `updateInstanceCondition` method does this for you, but since we now have two conditions, it makes more sense to make the deep copy ourselves.  If you look at any Kubernetes controller you'll find code that does the same thing.